enterprise/frontend/src/embedding-sdk/lib/plugins.ts

```diff
@@ -1,18 +1,39 @@
+import type { ReactNode } from "react";
+
+import type { DashCardMenuItem } from "metabase/dashboard/components/DashCard/DashCardMenu/DashCardMenu";
 import type { ClickAction, ClickObject } from "metabase/visualizations/types";
+import type Question from "metabase-lib/v1/Question";
 
 export type SdkDataPointObject = Pick<
   ClickObject,
   "value" | "column" | "data" | "event"
 >;
 
-export type SdkClickActionPluginsConfig = Pick<
-  SdkPluginsConfig,
-  "mapQuestionClickActions"
->;
+export type SdkClickActionPluginsConfig = (
+  clickActions: ClickAction[],
+  clickedDataPoint: SdkDataPointObject,
+) => ClickAction[];
+
+type DashCardMenuCustomElement = ({
+  question,
+}: {
+  question: Question;
+}) => ReactNode;
+
+export type DashCardCustomMenuItem = {
+  withDownloads?: boolean;
+  withEditLink?: boolean;
+  customItems?: (
+    | DashCardMenuItem
+    | (({ question }: { question?: Question }) => DashCardMenuItem)
+  )[];
+};
+
+export type SdkDashcardMenuPluginsConfig = {
+  dashcardMenu?: DashCardMenuCustomElement | DashCardCustomMenuItem;
+};
 
-export interface SdkPluginsConfig {
-  mapQuestionClickActions?: (
-    clickActions: ClickAction[],
-    clickedDataPoint: SdkDataPointObject,
-  ) => ClickAction[];
-}
+export type SdkPluginsConfig = {
```

Comment:
I inverted the types so that the "big" `SdkPluginsConfig` would be more reasonable to read, while the inner types would contain the actual typing rather than aliases